src/main/java/com/tabnine/statusBar/TabnineEnterpriseStatusBarWidget.java

```diff
@@ -0,0 +1,112 @@
+package com.tabnine.statusBar;
+
+import static com.tabnine.general.StaticConfig.LIMITATION_SYMBOL;
+import static com.tabnine.general.StaticConfig.getTabnineEnterpriseHost;
+
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.options.ShowSettingsUtil;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.popup.ListPopup;
+import com.intellij.openapi.wm.StatusBarWidget;
+import com.intellij.openapi.wm.impl.status.EditorBasedWidget;
+import com.intellij.util.Consumer;
+import com.tabnine.binary.requests.config.CloudConnectionHealthStatus;
+import com.tabnine.general.SubscriptionType;
+import com.tabnine.intellij.completions.LimitedSecletionsChangedNotifier;
+import com.tabnine.lifecycle.BinaryStateChangeNotifier;
+import com.tabnine.userSettings.AppSettingsConfigurable;
+import java.awt.event.MouseEvent;
+import javax.swing.*;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class TabnineEnterpriseStatusBarWidget extends EditorBasedWidget
```

Comment:
i separated the old widget class to 2 classes, and methodically took each "is onprem" condition and left the true case here and the false case in the old class.